lightning-invoice/src/utils.rs

```diff
@@ -71,8 +63,8 @@ pub fn create_phantom_invoice<Signer: Sign, K: Deref>(
 		invoice = invoice.amount_milli_satoshis(amt);
 	}
 
-	for (phantom_scid, real_node_pubkey, chan_details_list) in channels {
-		for channel in chan_details_list {
+	for hint in phantom_route_hints {
```

Comment:
Possibly would make these more difficult to filter, but I wonder if we should just have `get_phantom_route_hints` return `RouteHints`.

Reply:
Ah, nevermind. That would allow someone to call `create_phantom_invoice` with invalid hints.